src/toast-container.tsx

```diff
@@ -1,217 +1,310 @@
-   */
-  update = (
-    id: string,
-    message: string | JSX.Element,
-    toastOptions?: ToastOptions
-  ) => {
-    this.setState({
-      toasts: this.state.toasts.map((toast) =>
-        toast.id === id ? { ...toast, message, ...toastOptions } : toast
-      ),
-    });
-  };
-
-  /**
-   * Removes a toast from stack
-   */
-  hide = (id: string) => {
-    this.setState({
-      toasts: this.state.toasts.map((t) =>
-        t.id === id ? { ...t, open: false } : t
-      ),
-    });
-  };
-
-  /**
-   * Removes all toasts in stack
-   */
-  hideAll = () => {
-    this.setState({
-      toasts: this.state.toasts.map((t) => ({ ...t, open: false })),
-    });
-  };
-
-  /**
-   * Check if a toast is currently open
-   */
-  isOpen = (id: string) => {
-    return this.state.toasts.some((t) => t.id === id && t.open);
-  }
-
-  renderBottomToasts() {
-    const { toasts } = this.state;
-    let { offset, offsetBottom } = this.props;
-    let style: ViewStyle = {
-      bottom: offsetBottom || offset,
-      width: width,
-      justifyContent: "flex-end",
-      flexDirection: "column",
+        }
+
+        requestAnimationFrame(() => {
+            if(this.state.foldedToast === this.dummyToast) this.setState({ foldedToast: newToast }); 
+            else {
+                this.setState(
+                    {
+                        toastsHistory: [
+                            this.state.foldedToast,
+                            ...this.state.toastsHistory
+                        ],
+                        foldedToast: newToast
+                    }
+                )
+            }
+            });
+            return id;
+    };
+
+    renderUnfolded = () => {
+        this.setState({
+            unfolded: !this.state.unfolded
+        });
+    }
+
+    /**
+     * Updates a toast, To use this create you must pass an id to show method first, then pass it here to update the toast.
+     * @param id: string
+     * @param message: string | JSX.Element
+     * @param toastOptions?: ToastOptions
+     * @returns void
+     */
+    update = (
+        id: string,
+        message: string | JSX.Element,
+        toastOptions?: ToastOptions
+    ) => {
+        let msg: string | JSX.Element;
+        let options: ToastOptions;
+        if(this.state.foldedToast.id === id) {
+            msg = message !== this.state.foldedToast.message && message !== null ? message : this.state.foldedToast.message;
+            options = toastOptions !== this.state.foldedToast && toastOptions !== null ? {...this.state.foldedToast, ...toastOptions} : {...this.state.foldedToast};
+            debugger;
+            this.setState({ foldedToast: {...this.state.foldedToast, ...options, message: msg} });
+        }
+        else {
+            let foundToast = this.state.toastsHistory.find((t) => t.id === id);
+            // @ts-expect-error
+            msg = message !== undefined ? message : foundToast.message;
```

Comment:
message || foundToast.message

Takes left value, if left value is null, empty or undefined, takes right value
Attention: Empty string will also count as false

Reply:
This will make more sense if you look at the implementation of createGeneralNotification() returned functions. The comparison to null here is necessary

Reply:
<img width="1328" alt="Screenshot 2023-04-09 at 15 13 03" src="https://user-images.githubusercontent.com/55980542/230778165-7f7227ba-d1c0-4c39-886e-77ee189049a7.png">